### PR TITLE
 change Convert

### PR DESCRIPTION
**Pull request by VeronikaKorotkaia**

### CODE REVIEW

File: ConsoleApp2/Program.cs [hunk 1]
```diff
@@ -132,17 +132,17 @@ public static void BuyAnimal(PetShop petShop)
...
             Console.WriteLine("Please, enter animal name");
             string name = Console.ReadLine();
 
             Console.WriteLine("Please, enter animal price");
-            float price = (float)Convert.ToDouble(Console.ReadLine()); // Double привели к float
+            float price = EnterText(); 
 
             Console.WriteLine("Please, enter animal birthday");
-            DateTime birthday = DateTime.Parse(Console.ReadLine());
+            DateTime birthday = EnterData();
```
Comment (denys-prodan) [inline ConsoleApp2/Program.cs]: EnterData or EnterDate?

File: ConsoleApp2/Program.cs [hunk 2]
```diff
@@ -132,17 +132,17 @@ public static void BuyAnimal(PetShop petShop)
...
-            int type = Convert.ToInt32(Console.ReadLine());
+            int type = EnterNumber();
             var animalType = (AnimalType)type;
 
             Console.WriteLine("Please, enter animal name");
             string name = Console.ReadLine();
 
             Console.WriteLine("Please, enter animal price");
-            float price = (float)Convert.ToDouble(Console.ReadLine()); // Double привели к float
+            float price = EnterText();
```
Comment (denys-prodan) [inline ConsoleApp2/Program.cs]: Why EnterText?

File: ConsoleApp2/Program.cs [hunk 3]
```diff
@@ -160,10 +160,35 @@ public static int EnterNumber()
...
+        {
+            float value;
+            while (true)
+            {
+                var answer = Console.ReadLine();
+                if (float.TryParse(answer, out value))
+                {
+                    return value;
+                }
+                Console.WriteLine("Please, enter integer value");
```
Comment (denys-prodan) [inline ConsoleApp2/Program.cs]: It asks to enter "integer" value

File: ConsoleApp2/Program.cs [hunk 4]
```diff
@@ -160,10 +160,35 @@ public static int EnterNumber()
...
+        {
+            DateTime value;
+            while(true)
+            {
+                var answer = Console.ReadLine();
+                if (DateTime.TryParse(answer, out value))
+                {
+                    return value;
+                }
+                Console.WriteLine("Please, enter integer value");
```
Comment (denys-prodan) [inline ConsoleApp2/Program.cs]: Should be "date"?

**Review by denys-prodan — CHANGES_REQUESTED** [review]

File: ConsoleApp2/Program.cs (hunk 1 above)
Comment (VeronikaKorotkaia) [reply ConsoleApp2/Program.cs]: EnterDate

File: ConsoleApp2/Program.cs (hunk 2 above)
Comment (VeronikaKorotkaia) [reply ConsoleApp2/Program.cs]: EnterPrice?

File: ConsoleApp2/Program.cs (hunk 4 above)
Comment (VeronikaKorotkaia) [reply ConsoleApp2/Program.cs]: ok

File: ConsoleApp2/Program.cs (hunk 3 above)
Comment (VeronikaKorotkaia) [reply ConsoleApp2/Program.cs]: i don't understand

Comment (denys-prodan) [reply ConsoleApp2/Program.cs]: You need to enter float value. But asking user to enter "integer".

File: ConsoleApp2/Program.cs (hunk 2 above)
Comment (denys-prodan) [reply ConsoleApp2/Program.cs]: EnterFloat

**Pull request merged**